File: angular_frontend/src/app/farmer.ts

export class Farmer {
    id: number = 0;
    name: string = '';
    phoneNumber: string = '';


    constructor(
        id?: number,
        name?: string,
        phoneNumber?: string
    ) {
        if (id) this.id = id;
        if (name) this.name = name;
        if (phoneNumber) this.phoneNumber = phoneNumber;
    }
}

File: farmerSevice/src/main/java/com/example/FarmerService/entity/Farmer.java
package com.example.FarmerService.entity;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder

@Table(name = "farmers")
public class Farmer implements Serializable {
        private static final long serialVersionUID = 1L;
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Long id;

        @Column(nullable = false)
        private String fullName;

        @Column(nullable = false, unique = true)
        private String nationalId;

        private String phoneNumber;

        private String email;

        private String location;

        @Column(name = "created_at", updatable = false)
        private LocalDateTime createdAt;

        @Column(name = "updated_at")
        private LocalDateTime updatedAt;

        @PrePersist
        protected void onCreate() {
            this.createdAt = LocalDateTime.now();
        }

        @PreUpdate
        protected void onUpdate() {
            this.updatedAt = LocalDateTime.now();
        }

        // Getters and setters
    }


